webv2/typemap/postgres_typemap.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typemap
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+)
+
+// toSpannerTypePostgres defines the mapping of source types into Spanner
+// types. Each source type has a default Spanner type, as well as other potential
+// Spanner types it could map to. When calling toSpannerTypePostgres, you specify
+// the source type name (along with any modifiers), and optionally you specify
+// a target Spanner type name (empty string if you don't have one). If the target
+// Spanner type name is specified and is a potential mapping for this source type,
+// then it will be used to build the returned ddl.Type. If not, the default
+// Spanner type for this source type will be used.
+// Note that toSpannerTypePostgres is extensively tested via tests in web_test.go.
+//
+// TODO: Move the type remapping function to toddl.go (once we've merged
+// dynamodb/toddl.go, mysql/toddl.go and postgres/toddl.go).
+// Consider some refactoring to reduce code duplication (although note
+// that this type remapping has to preserve all previous changes done via the UI!)
+func ToSpannerTypePostgres(srcType string, spType string, mods []int64) (ddl.Type, []internal.SchemaIssue) {
```

Comment:
can we not use the toddl.go toSpannerTypeInternal method instead of these ?

Reply:
also remove these files from existing code

Reply:
Their implementation is somewhat different. And these files(postgres_typemap.go,mysql_typemap.go and sqlserver_typemap.go) were already present in root of webv2 folder and not a new addition for this milestone. We just move them inside webv2/typemap folder as a refactoring process.

Reply:
I understand that they are moved because of refactoring but since the initial Pr was huge these things might have missed out. Therefore, if possible we should remove the duplication in code

Reply:
Yes I agree that there is code duplication. It looks like typemap in webv2 are superset of what is there in internal. So we may need to replace the typemap of internal with the typemap of webv2 right.